README.md

```diff
@@ -16,10 +16,199 @@ Alternatives for PHP:
+    * [Usage](#usage)
+        * [Performing login](#performing-login)
+        * [Performing logout](#performing-logout)
+        * [Complete API](#complete-api)
+    * [Example](#example)
+        * [Demo application](#demo-application)
+* [Features](#features)
+    * [More features](#more-features)
+* [Troubleshooting](#troubleshooting)
+* [Testing](#testing)
+    * [Unit Tests](#unit-tests)
+    * [Linting](#linting)
+* [Contributing](#contributing)
+* [See also](#see-also)
+* [Authors](#authors)
+* [License](#license)
+
+
+## Repository layout
+
+* [bin/](bin/) auxiliary scripts
+* [example/](example/) contains a demo application
+* [src/](src/) contains the library implementation
+* [test/](test/) contains the unit tests
+
+## Getting Started
+
+Tested on: amd64 Debian 9.5 (stretch, current stable) with PHP 7.0.
+
+Supports PHP 7.0, 7.1 and 7.2.
+
+### Prerequisites
+
+```sh
+sudo apt install composer make openssl php-curl php-zip php-xml phpunit
+```
+
+### Configuring and Installing
+
+
+**NOTE**: during testing, please use the test Identity Provider [spid-testenv2](https://github.com/italia/spid-testenv2).
+
+
+1. Install with composer 
+    
+    ```composer require italia/spid-php-lib```
+
+2. Generate key and certificate files for your Service Provider (SP).
+
+    Example: 
+    ```openssl req -x509 -nodes -sha256 -days 365 -newkey rsa:2048 -subj "/C=IT/ST=Italy/L=Milan/O=myservice/CN=localhost" -keyout sp.key -out sp.crt & wait;\```
+
+3. Download the Identity Provider (IdP) metadata files and place them in a directory in your project, for example `idp_metadata`. 
+    A convenience tool is provided to download those of the production IdPs: [bin/download_idp_metadata.php](bin/download_idp_metadata.php), example usage:
+    ```sh
+    bin/download_idp_metadata.php ./example/idp_metadata
+    ```
+    
+    *TEST ENVIRONMENT: If you are using [spid-testenv2](https://github.com/italia/spid-testenv2), manually download the IDP metadata and place it in your `idp_metadata` folder*
+
+4. Make your SP known to IDPs: for production follow the guidelines at [https://www.spid.gov.it/come-diventare-fornitore-di-servizi-pubblici-e-privati-con-spid](https://www.spid.gov.it/come-diventare-fornitore-di-servizi-pubblici-e-privati-con-spid)
+
+    *TEST ENVIRONMENT: simply download your Service Provider (SP) metadata and place it in the appropriate folder of the [test environment](https://github.com/italia/spid-testenv2). The test environment must be restarted.*
+
+
+
+### Usage
+
+All classes provided by this package reside in the `Italia\Spid` namespace.
+
+Load them using the composer-generated autoloader:
+```php
+require_once(__DIR__ . "/../vendor/autoload.php");
+```
+
+The main class is `Italia\Spid\Sp` (service provider).
+Generate a settings array following this guideline
+
+```php
+$settings = array(
+    'sp_entityid' => SP_BASE_URL, // Example: https://example.com/my-sp-service
```

Comment:
usare qui lo stesso protocollo/fqdn che usiamo come default in example (https://sp.example.com) ?